Unreal/DazStudioPlugin/DzUnrealAction.cpp

```diff
@@ -56,51 +56,53 @@ void DzUnrealAction::executeAction()
 	 // Create and show the dialog. If the user cancels, exit early,
 	 // otherwise continue on and do the thing that required modal
 	 // input from the user.
-
-	 if (dzScene->getNumSelectedNodes() != 1)
-	 {
-		  QMessageBox::warning(0, tr("Error"),
-				tr("Please select one Character or Prop to send."), QMessageBox::Ok);
-		  return;
-	 }
-
-	 // Create the dialog
-	 DzUnrealDialog* dlg = new DzUnrealDialog(mw);
-
-	 // If the Accept button was pressed, start the export
-	 if (dlg->exec() == QDialog::Accepted)
-	 {
-		 // Collect the values from the dialog fields
-		 CharacterName = dlg->assetNameEdit->text();
-		 ImportFolder = dlg->intermediateFolderEdit->text();
-		 CharacterFolder = ImportFolder + "\\" + CharacterName + "\\";
-		 CharacterFBX = CharacterFolder + CharacterName + ".fbx";
-		 CharacterBaseFBX = CharacterFolder + CharacterName + "_base.fbx";
-		 CharacterHDFBX = CharacterFolder + CharacterName + "_HD.fbx";
-		 AssetType = dlg->assetTypeCombo->currentText().replace(" ", "");
-		 MorphString = dlg->GetMorphString();
-		 Port = dlg->portEdit->text().toInt();
-		 ExportMorphs = dlg->morphsEnabledCheckBox->isChecked();
-		 ExportSubdivisions = dlg->subdivisionEnabledCheckBox->isChecked();
-		 MorphMapping = dlg->GetMorphMapping();
-		 ShowFbxDialog = dlg->showFbxDialogCheckBox->isChecked();
-		 ExportMaterialPropertiesCSV = dlg->exportMaterialPropertyCSVCheckBox->isChecked();
-		 SubdivisionDialog = DzUnrealSubdivisionDialog::Get(dlg);
-		 FBXVersion = dlg->fbxVersionCombo->currentText();
-
-		 if (AssetType == "SkeletalMesh" && ExportSubdivisions)
-		 {
-			 // export base mesh
-			 ExportBaseMesh = true;
-			 SubdivisionDialog->LockSubdivisionProperties(false);
-			 Export();
-		 }
-
-		 ExportBaseMesh = false;
-		 SubdivisionDialog->LockSubdivisionProperties(ExportSubdivisions);
-		 Export();
-
-	 }
+     if (getNonInteractiveMode() > 0) {
```

Comment:
Clarifications:

1. When NonInteractiveMode == 0, program should proceed normally and display all modal dialog boxes that require user interaction, just like before the update.
2. When NonInteractiveMode ==1, program should ALSO proceed with all the instruction calls, but it should SKIP anything that requires user interaction.  There are two things that require user interaction that should be skipped in this function: 
           - QMessageBox() functions will require user to click OK to continue, so they should be skipped
           - dlg->exec() returns QDialog::Accepted when user clicks OK and QDialog::Rejected when user clicks CANCEL
                      - If NonInteractiveMode == 1, treat as automatic OK and continue with further processing just like original code.
                      - If NonInteractiveMode != 0 and != 1, then treat as automatic cancel and skip further processing.